### PR TITLE
Move block animations from block_svg.js  to a new file named block_animations.js

### PR DESCRIPTION
**Pull request by gamefroot-developer**
RE: Issue #1022

The following functions in block_svg.js have been moved to a new file named block_animations.js:

Blockly.BlockSvg.connectionUiStep_, disposeUiStep_, disconnectUiStep_, disconnectUiStop_, disconnectUiStop_.pid, and disconnectUiStop_.group

Regards
@gamefroot-developer

### CODE REVIEW

File: core/block_animations.js [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+ * Not normally called directly, workspace.newBlock() is preferred.
+ * @param {!Blockly.Workspace} workspace The block's workspace.
+ * @param {?string} prototypeName Name of the language object containing
+ *     type-specific functions for this block.
+ * @param {string=} opt_id Optional ID.  Use this ID if provided, otherwise
+ *     create a new id.
+ * @extends {Blockly.Block}
+ * @constructor
+ */
+Blockly.BlockAnimations = function(workspace, prototypeName, opt_id) {
```
Comment (rachel-fenichel) [inline core/block_animations.js]: It looks like you copied this constructor without removing old code.  Blockly.BlockAnimations shouldn't have a constructor.  Please remove this function and its comment.

File: core/block_animations.js [hunk 2]
```diff
@@ -0,0 +1,181 @@
+/**
+ * @license
+ * Visual Blocks Editor
+ *
+ * Copyright 2012 Google Inc.
```
Comment (rachel-fenichel) [inline core/block_animations.js]: Update date

File: core/block_animations.js [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Methods for graphically rendering a block as SVG.
```
Comment (rachel-fenichel) [inline core/block_animations.js]: Update fileoverview and author

File: core/block_animations.js [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+ * @author fraser@google.com (Neil Fraser)
+ */
+'use strict';
+
+goog.provide('Blockly.BlockAnimations');
```
Comment (rachel-fenichel) [inline core/block_animations.js]: Add
```
/**
 * @name Blockly.BlockAnimations
 * @namespace
 **/
```

File: core/block_animations.js [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+  this.svgPath_.tooltip = this;
+
+  /** @type {boolean} */
+  this.rendered = false;
+
+  Blockly.Tooltip.bindMouseEvents(this.svgPath_);
+  Blockly.BlockAnimations.superClass_.constructor.call(this,
+      workspace, prototypeName, opt_id);
+};
+goog.inherits(Blockly.BlockAnimations, Blockly.Block);
```
Comment (rachel-fenichel) [inline core/block_animations.js]: No inheritance.  BlockAnimations is just a namespace, not a full class.

File: core/block_animations.js [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+    group.setAttribute('transform', group.translate_);
+    Blockly.BlockAnimations.disconnectUiStop_.group = null;
+  }
+};
+
+/**
+ * PID of disconnect UI animation.  There can only be one at a time.
+ * @type {number}
+ */
+Blockly.BlockAnimations.disconnectUiStop_.pid = 0;
```
Comment (rachel-fenichel) [inline core/block_animations.js]: Move this to the top of the file, just after the calls to `goog.require`

File: core/block_animations.js [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+
+/**
+ * SVG group of wobbling block.  There can only be one at a time.
+ * @type {Element}
+ */
+Blockly.BlockAnimations.disconnectUiStop_.group = null;
```
Comment (rachel-fenichel) [inline core/block_animations.js]: Also move this to the top of the file.

File: core/block_svg.js [hunk 8]
```diff
@@ -31,6 +31,7 @@ goog.require('Blockly.ContextMenu');
 goog.require('Blockly.RenderedConnection');
 goog.require('Blockly.Touch');
 goog.require('Blockly.utils');
+goog.require('Blockly.BlockAnimations');
```
Comment (rachel-fenichel) [inline core/block_svg.js]: Move just below `goog.require('Blockly.Block');`
These imports are in alphabetical order.

**Comment by Razaik25** [comment]
@rachel-fenichel No we are not connected. Sorry, I got busy. I can submit the pull request in a day.

**Comment by rachel-fenichel** [comment]
@Razaik25 @gamefroot-developer I'll let you all sort this out, and I'll review code as I get it.

Cheers,
Rachel

**Comment by rachel-fenichel** [comment]
I'm closing this PR due to a lack of activity.

**Pull request closed without merging**